trl/trainer/reward_trainer.py

```diff
@@ -159,6 +160,19 @@ def __init__(
 
                     model = prepare_model_for_kbit_training(model, **prepare_model_kwargs)
 
+                # Warn if the user passes "all-linear" for the target_modules
+                target_modules = (
+                    peft_config.get("target_modules", None)
+                    if isinstance(peft_config, dict)
+                    else peft_config.target_modules
+                )
```

Comment:
I am handling both cases of `peft_config` being `dict` and being a `PeftConfig`object. 

It is type-hinted as a `dict` in the `RewardModel` docstring, but I can see a `PeftConfig` object is passed to `RewardTrainer` [here](https://github.com/huggingface/trl/blob/bbdd6db17c49db813695d0a8bc0da7bf6b1bb88e/examples/scripts/reward_modeling.py#L119).